### PR TITLE
Rel constants script

### PR DESCRIPTION
**Pull request by Crell**
Replaces #5.

Here's a new version that uses the published IANA link registry to build a complete set of constants.

Bizzarely, IANA has scripts blocked from downloading the XML file.  That is moronic, IMO, but c'est la vie.  I've checked it in here, but the whole thing can be rebuilt by updating that file with wget and rerunning the build script, then committing everything.

There's some odd spacing in the generated code docblocks due to not all relations having a `note`, but... for generated code I honestly don't care.

Also, the generated code should be PHP 5 compatible still.  The build script isn't, since that's new code and basically no one but maintainers would be running it.  When this package gets updated to the 2.0 version of the interface I'll also bump it to use PHP 7 visibility on the constants, as well as PHP 7-ify the rest of the package.

### CODE REVIEW

**Comment by evert** [comment]
neat!

**Review by dunglas — APPROVED** [review]
Nice!

**Comment by weierophinney** [comment]
> Bizzarely, IANA has scripts blocked from downloading the XML file. That is moronic, IMO, but c'est la vie. I've checked it in here, but the whole thing can be rebuilt by updating that file with wget and rerunning the build script, then committing everything.

IIRC, the way we worked around this was to provide a `User-Agent` header with something mimicing a browser user-agent string.

File: rebuild-rels.php [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        $description = $this->rewrap((string)$record->description);
+        $note = $this->rewrap((string)$record->note);
+        $seeUri = $this->xrefLink($record->spec->xref);
+
+        return <<<END
+
+    /**
+     * {$description}
+     *
+     * {$note}
```
Comment (weierophinney) [inline rebuild-rels.php:103]: The majority of the entries do not have notes, which means we end up with the majority having structures like this:

```php
/**
  * Refers to...
  *
  *
  *
  * @see iana link
  */
```

In other words, two extra lines for each docblock.

I'm thinking that maybe this should be:

```php
/**
  * {$description}{$note}
  *
  * @see {$seeUri}
  */
```

If a note is detected, then it uses the following format:

```php
private const NOTE_TEMPLATE = <<< 'END'

 * {$note}
 *
END;
```

Yes, a bit more work, but it makes the source more readable in the end.

**Review by weierophinney — CHANGES_REQUESTED** [review]
Two comments:

- You _should_ be able to download the script from the IANA if you pass a `User-Agent` header. I just tried the following: `curl_setopt($handle, CURLOPT_HTTPHEADER, ['User-Agent; php-fig/fig-utils'])` and had success.
- Let's omit the `$note` portion of the generated docblocks if there is no note.

File: rebuild-rels.php [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+/**
+ * Compiles a list of SimpleXmlElements to a Relations interface.
+ */
+class RegistryCompiler
+{
+    /**
+     * Writes a Relations index class based on the provided records.
+     *
+     * @param iterable<\SimpleXMLElement> $records
+     *   An itrable of SimpleXml elements from the Link Relations XML file.
```
Comment (weierophinney) [inline rebuild-rels.php]:
```suggestion
     *   An iterable of SimpleXml elements from the Link Relations XML file.
```

**Review by weierophinney — APPROVED** [review]
One minor typo, but otherwise looks good. :ship:

**Pull request merged**